pkg/network/discovery.go

```diff
@@ -79,6 +79,7 @@ func (d *DefaultDiscovery) PoolCount() int {
 func (d *DefaultDiscovery) pushToPoolOrDrop(addr string) {
 	select {
 	case d.pool <- addr:
+		updatePoolCountMetric(d.PoolCount())
```

Comment:
There is a case of removing elements from the pool, metric should be updated there also.